internal/controllers/clusterextension_controller.go

```diff
@@ -568,10 +610,22 @@ func clusterExtensionRequestsForCatalog(c client.Reader, logger logr.Logger) crh
 	}
 }
 
-func (r *ClusterExtensionReconciler) resolve(ctx context.Context, clusterExtension ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, error) {
+type ResolveErrorType int
+
+const (
+	PackageNotFoundError ResolveErrorType = iota
+	VersionNotFoundError
+	OtherResolveError
+)
+
+func (r ResolveErrorType) String() string {
+	return [...]string{"PackageNotFound", "VersionNotFound", "Other"}[r]
+}
+
+func (r *ClusterExtensionReconciler) resolve(ctx context.Context, clusterExtension ocv1alpha1.ClusterExtension) (*catalogmetadata.Bundle, ResolveErrorType, error) {
```

Comment:
Why are we adding another return value that always ends up being the same value? in other words, it does not appear that `PackageNotFoundError` nor "VersionNotFoundError` are used.
Shouldn't these be rolled up into the `error` return?
Is this to avoid aggregate errors? If so, there's got to be a better way.

Reply:
i.e. the linter is complaining:
```
(*ClusterExtensionReconciler).resolve - result 1 (github.com/operator-framework/operator-controller/internal/controllers.ResolveErrorType) is always 2 (unparam)
```

Reply:
So, the linter is still complaining about this. And I'm not sure this is the right solution, here. It also seems that you are rewriting `resolve()` because you moved it further down in the file - diff is having difficulties. Can you put it back where it was originally?

Reply:
added return value removed.